src/main/java/org/terasology/module/inventory/systems/StartingInventorySystem.java

```diff
@@ -65,6 +66,15 @@ public void onStartingInventory(OnPlayerSpawnedEvent event,
         logErrors();
     }
 
+    @ReceiveEvent
+    public void onRequestInventory(RequestInventoryEvent event,
+                                   EntityRef player,
+                                   StartingInventoryComponent startingInventory) {
+        entitiesWithoutInventory.clear();
+        addItemsTo(startingInventory.items, player, player.getParentPrefab().getName());
+        logErrors();
+    }
```

Comment:
This introduces some indirect coupling from the `RequestInventoryEvent` the `StartingInventory`  component, which might be hard to understand in the future.

Instead of adding state to the entity (in form of the starting inventory component) and then sending an empty trigger event to read data from that component, it might be more straight forward to just have the payload in the event itself. 

Also, note that the entity needs to have an `InventoryComponent` for this to work at all. Thus, I think it would be better to go more towards the following:

```suggestion
    @ReceiveEvent(InventoryComponent.class)
    public void onRequestInventory(RequestInventoryEvent event, EntityRef player) {
        entitiesWithoutInventory.clear();
        addItemsTo(event.items, player, player.getParentPrefab().getName());
        logErrors();
    }
```